clusterman/kubernetes/kubernetes_cluster_connector.py

```diff
@@ -184,6 +204,44 @@ def uncordon_node(self, node_name: str) -> bool:
             logger.warning(f"Failed to uncordon {node_name}: {e}")
             return False
 
+    def list_node_migration_resources(self, statuses: List[MigrationStatus]) -> Collection[MigrationEvent]:
+        """Fetch node migration event resource from k8s CRD
+
+        :param List[MigrationStatus] statuses: event status to look for
+        :return: collection of migration events
+        """
+        assert self._migration_crd_api, "CRD client was not initialized"
+        try:
+            label_filter = ",".join(status.value for status in statuses)
+            resources = self._migration_crd_api.list_cluster_custom_object(
+                label_selector=f"{MIGRATION_CRD_STATUS_LABEL} in ({label_filter})",
+            )
+            return set(map(MigrationEvent.from_crd, resources.get("items", [])))
+        except Exception as e:
+            logger.error(f"Failed fetching migration events: {e}")
+        return set()
+
+    def mark_node_migration_resource(self, event_name: str, status: MigrationStatus) -> None:
+        """Set status for migration event resource
+
+        :param str event_name: name of the resource CRD
+        :status MigrationStatus status: status to be set
+        """
+        assert self._migration_crd_api, "CRD client was not initialized"
+        try:
+            self._migration_crd_api.patch_cluster_custom_object(
+                name=event_name,
+                body={
+                    "metadata": {
+                        "labels": {
+                            MIGRATION_CRD_STATUS_LABEL: status.value,
```

Comment:
I think it will be better keep labels immutable. If we want to patch/edit some fields, we should use spec.{field}.
What do you think store status in spec instead of label?

Reply:
Keeping the status in labels allows querying the data much more easily. This was discussed with robj as well on the tech-spec review.